File: flutter/lib/domain/use_case/flashcard/update_flashcard_use_case.dart
import '../../entities/flashcard.dart';
import '../../repository/flashcard_repository.dart';

class UpdateFlashcardUseCase {
  final FlashcardRepository _flashcardRepository;

  UpdateFlashcardUseCase(this._flashcardRepository);

  Future<void> call(Flashcard updatedFlashcard) async {
    await _flashcardRepository.updateFlashcard(updatedFlashcard);
  }
}

File: android/app/src/main/java/com/synergyboat/flashcardAi/domain/usecase/flashcard/UpdateFlashcardUseCase.kt
package com.synergyboat.flashcardAi.domain.usecase.flashcard

import com.synergyboat.flashcardAi.domain.entities.Flashcard
import com.synergyboat.flashcardAi.domain.repository.FlashcardRepository

class UpdateFlashcardUseCase(
    private val repository: FlashcardRepository
) {
    suspend operator fun invoke(updatedFlashcard: Flashcard) {
        repository.updateFlashcard(updatedFlashcard)
    }
}